src/session/content/item_row.rs

```diff
@@ -94,16 +98,290 @@ impl ItemRow {
+                                ));
+                            }
+                            MessageContent::MessageChatJoinByRequest => {
+                                self.get_or_create_event_row().set_label(&format!(
+                                    "<b>{}</b>",
+                                    if message.is_outgoing() {
+                                        gettext("You joined the group")
+                                    } else {
+                                        gettext!("{} joined the group", sender_name)
+                                    }
+                                ));
+                            }
+                            MessageContent::MessageChatJoinByLink => {
+                                self.get_or_create_event_row().set_label(&format!(
+                                    "<b>{}</b>",
+                                    if message.is_outgoing() {
+                                        gettext("You joined the group via invite link")
+                                    } else {
+                                        gettext!("{} joined the group via invite link", sender_name)
+                                    }
+                                ));
+                            }
+                            MessageContent::MessageChatAddMembers(data) => {
+                                let my_user_id = message.chat().session().me().id();
+                                self.get_or_create_event_row().set_label(&format!(
+                                    "<b>{}</b>",
+                                    if message.sender().as_user().map(User::id).as_ref()
+                                        == data.member_user_ids.get(0)
+                                    {
+                                        if message.is_outgoing() {
+                                            gettext("You joined the group")
+                                        } else {
+                                            gettext!("{} joined the group", sender_name)
+                                        }
+                                    } else {
+                                        let members =
+                                            remove_you(data.clone().member_user_ids, my_user_id)
+                                                .into_iter()
+                                                .map(|user_id| {
+                                                    message
+                                                        .chat()
+                                                        .session()
+                                                        .user_list()
+                                                        .get(user_id)
+                                                })
+                                                .map(|user| self::sender_name(&user))
+                                                .collect::<Vec<_>>();
```

Comment:
Here are some issues.

1. The function should rather be named `remove_me` (but we don't need this function anyway)
2. Within the function `remove_you` you make an unnecessary clone at the end while the `Vec<_>` could just be moved out.
2. You clone the whole `data` while you only want to have `member_user_ids`. So the `.clone()` should be behind that.
3. In every `map` you retrieve the user list from the session, but it could be retrieved beforehand.

Here is my suggestion
```suggestion
                                        let session = message.chat().session();
                                        let user_list = session.user_list();
                                        let members = data
                                            .member_user_ids
                                            .iter()
                                            .copied()
                                            .filter(|user_id| *user_id != my_user_id)
                                            .map(|user_id| user_list.get(user_id))
                                            .map(|user| self::sender_name(&user))
                                            .collect::<Vec<_>>();
```